components/service/glean/src/main/java/mozilla/components/service/glean/net/HttpPingUploader.kt

```diff
@@ -33,17 +32,17 @@ internal class HttpPingUploader(configuration: Configuration) : PingUploader {
      * @param data the serialized text data to send
      */
     private fun logPing(path: String, data: String) {
-        if (config.logPings) {
+        if (Glean.configuration.logPings == true) {
```

Comment:
Currently you are copying the values over; so I assume you could still use the passed in configuration? Anyhow, instead of accessing the configuration via the singleton (which is awkward in tests etc.) could we pass the configuration that should be used to `upload()`? After all that's the part that requires a configuration.

Reply:
This is a good point. I'm now changing the configuration object as you suggested early on in the review, so it's ok to pass the `configuration` object to upload :)